rules/LeagueEvent/Rector/MethodCall/DispatchStringToObjectRector.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Rector\LeagueEvent\Rector\MethodCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\New_;
+use PhpParser\Node\Name\FullyQualified;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Return_;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\StringType;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\LeagueEvent\Rector\MethodCall\DispatchStringToObjectRector\DispatchStringToObjectRectorTest
+ */
+final class DispatchStringToObjectRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Change string events to objects which implement \League\Event\HasEventName', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    /** @var \League\Event\EventDispatcher */
+    private $dispatcher;
+
+    public function run()
+    {
+        $this->dispatcher->dispatch('my-event');
+    }
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    /** @var \League\Event\EventDispatcher */
+    private $dispatcher;
+
+    public function run()
+    {
+        $this->dispatcher->dispatch(new class implements \League\Event\HasEventName
```

Comment:
This seems very confusing. Could you link original upgrade source that recommend this?

It should be at least own variable...

```suggestion
-        $this->dispatcher->dispatch(new class implements \League\Event\HasEventName
+        $myEvent = new class implements \League\Event\HasEventName 
+        $this->dispatcher->dispatch($myEvent);
```

...or better standalone event class, as the event can be dispatched used on multiple places and it anonymous class makes event class duplicated per each instance. This is standard way in Symfony and Nette, that allow argument reusability and validation via value object ctor.

Reply:
There is no recommendation in original package, I just fall into this when upgrading one of my projects. Old version used to support string events. New doesn’t. I found this solution as the easiest way how to upgrade package without additional configuration. Of course the best way would be create some standalone event class, but what will be its name? What namespace will it use? Where it will be placed? Does Rector support creating new file at all?

Reply:
There is quite similar rule for converting magic Nette `on*` properties to Symfony-like event classes:

https://github.com/rectorphp/rector-nette/tree/main/tests/Kdyby/Rector/MethodCall/ReplaceMagicPropertyEventWithEventClassRector

Reply:
Thanks, I went through this and I still believe that anonymous class is the smartest and simplest solution for this (not very used use case). If somebody will want to create standalone class, he will create it before/after running rector. This anonymous class will not create new class(es) in the target project and will work like old version.

Reply:
> It should be at least own variable...
> 
Can I change parent node in rector? If I want to create variable $myEvent before method call, like you suggested, I cannot modify method call but some other type of node right? Again, it will overcomplicate this simple solution